src/administration/Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs

```diff
@@ -320,7 +342,20 @@ public async Task DeleteConnectorAsync(Guid connectorId, Guid userId, Cancellati
             con.LastEditorId = userId;
             con.DateLastChanged = DateTimeOffset.UtcNow;
         });
-        await _dapsService.DeleteDapsClient(result.DapsClientId, cancellationToken).ConfigureAwait(false);
+        await _dapsService.DeleteDapsClient(dapsClientId, cancellationToken).ConfigureAwait(false);
+        await _portalRepositories.SaveAsync();
+    }
+
+    private async Task DeleteConnectorWithDocuments(Guid connectorId, Guid selfDescriptionDocumentId, IConnectorsRepository connectorsRepository)
+    {
+        _portalRepositories.GetInstance<IDocumentRepository>().RemoveDocument(selfDescriptionDocumentId!);
```

Comment:
The `!` can be removed since the parameter isn't nullable

```suggestion
        _portalRepositories.GetInstance<IDocumentRepository>().RemoveDocument(selfDescriptionDocumentId);
```